Solution UI: fix components list refresh management

carlito-scality opened the pull request:
**Component**:Solution UI

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 
- When we switch from environment 1 to environment 2, there are 2 refresh of clockServer running instead of one in the environment 2 since the refresh of the environment 1 is not cancelled correctly. The root cause is the redux state "isRefreshing" in examples/metalk8s-solution-example/ui/src/ducks/app/clockServer.js is shared for all environments. It should be part of one single environment

**Summary**:

**Acceptance criteria**: 
- No regression
- When switching from one environment to another one, the component refreshs are correctly managed

---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: https://github.com/scality/metalk8s/issues/1820

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

carlito-scality committed:
Solution UI: move clockServer and versionServer redux state in environment state to manage correctly their refreshs

carlito-scality committed:
Merge branch 'development/2.4' into bugfix/fix-conponents-list-refresh

* development/2.4: (7 commits)
  salt: Add saltenv when running highstate
  ...

bert-e commented:
# Hello carlito-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

carlito-scality committed:
Rename clockServer and versionServer FETCH methods
- We put them in plurals fetchClockServers/fetchVersionServers) to list components because we will have fetchClockServer and fetchVersionServer to read a perticular component

carlito-scality committed:
Solution UI: fetch component when editing it
- When arriving on component edit page, we have to fetch the component information to fill out the form and update  its state, otherwise the form is blank.

carlito-scality commented:
@bert-e approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/4128796787f8cac7c5b9bba3ab4eed801e42baef) did not succeed in branch bugfix/fix-conponents-list-refresh.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye carlito-scality.
